sandbox/src/main/java/org/robolectric/internal/bytecode/ShadowWrangler.java

```diff
@@ -305,9 +306,18 @@ private Method findShadowMethod(
       Class<?>[] types,
       ShadowInfo shadowInfo,
       Class<?> shadowClass) {
+    // First, try to find shadow method with the exact method signature.
     Method method = findShadowMethodDeclaredOnClass(shadowClass, name, types);
 
+    if (method == null) {
+      // Second, try to find mapped shadow method with the exact method signature except method
+      // name.
+      method = findMappedShadowMethodDeclaredOnClass(shadowClass, name, types);
```

Comment:
There may be some performance implications of this, though I would like to measure it first. A rule of thumb is 90-95% of Android methods that get called do not have corresponding shadow methods. With this change, there will be extra reflection work done for 90-95% of these method calls. Maybe it's fine, since shadows typically don't contain too many methods. But I would like to quantify the impact of this change (I can get an estimate).

Reply:
An alternative may be to use the Processor to generate a list of shadow methods that are renamed, though that would not work for custom shadows.

Reply:
The idea would be store a map in ShadowInfo that contains these renamed methods.